### ISSUE TITLE
code inside `TYPE_CHECKING` blocks should be treated like stubs

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ty#604]
```py
from typing import TYPE_CHECKING

if TYPE_CHECKING:
   def foo() -> int: ... # Function always implicitly returns `None`, which is not assignable to return type `int` (invalid-return-type)
```
https://play.ty.dev/cd6ee5d1-98e2-48b8-be7c-6bfcda4bbbd1

there's no point adding an implementation here to fix the error because it doesn't actually get executed at runtime, so it would be nice if ty treated it the same as a `.pyi` file
